Classes/Job/JobManager.php

```diff
@@ -175,6 +176,18 @@ public function peek(string $queueName, int $limit = 1): array
         }, $messages);
     }
 
+    /**
+     * This method is here to be called by queues if they reached local polling timeouts, if this applies
+     *
+     * @throws InterruptException
```

Comment:
Does it really throw this exception? (I'm not so much into process control signals as you can tell)

Reply:
Yeah. It calls the signal handler - which might trigger the exception, if the signal has been received. No chance to annotate this in any other way - at least none I am aware of.
The exception would by thrown by the registered signal handler in `workCommand`, if it had been received meanwhile

Reply:
Sure, makes sense of course!